pkg/server/api/svid/v1/service.go

```diff
@@ -324,6 +347,64 @@ func (s *Service) NewJWTSVID(ctx context.Context, req *svid.NewJWTSVIDRequest) (
 	}, nil
 }
 
-func (s *Service) NewDownstreamX509CA(context.Context, *svid.NewDownstreamX509CARequest) (*svid.NewDownstreamX509CAResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+func (s *Service) NewDownstreamX509CA(ctx context.Context, req *svid.NewDownstreamX509CARequest) (*svid.NewDownstreamX509CAResponse, error) {
+	log := rpccontext.Logger(ctx)
+
+	if err := rpccontext.RateLimit(ctx, 1); err != nil {
+		log.WithError(err).Error("Rejecting request due to downstream CA signing rate limit")
+		return nil, err
+	}
+
+	downstreamEntries, isDownstream := rpccontext.CallerDownstreamEntries(ctx)
+	if !isDownstream || downstreamEntries == nil || len(downstreamEntries) == 0 {
+		// The real CallerDownstreamEntries will never return empty, but it helps for writing unit tests
+		err := status.Error(codes.NotFound, "error not found or not authorized")
+		log.WithError(err).Error("Rejecting NewDownstreamX509CA request because caller has no downstream entries")
+		return nil, err
```

Comment:
Either way is fine with me. I don't see any harm in unnecessary checks but I don't want to clutter things.

Reply:
>I think I'd lean towards just asserting that isDownstream is true, since that would be a misconfiguration. Having no entries would mean a catastrophic bug in the middleware in violation of the contract.

I'm OK just asserting that `isDownstream` is true. Adding more checks seems to be unnecessary, adding no real benefit (at the cost of more code to maintain).